source/PlayerInfo.cpp

```diff
@@ -683,78 +683,45 @@ const Date &PlayerInfo::GetDate() const
+	if(!amount)
+		return;
+	else if(amount > 0)
 	{
-		if(date < it->GetDate())
-			++it;
-		else
+		while(amount--)
 		{
-			it->Apply(*this);
-			it = gameEvents.erase(it);
-		}
-	}
+			++date;
 
-	// Check if any missions have failed because of deadlines and
-	// do any daily mission actions for those that have not failed.
-	for(Mission &mission : missions)
-	{
-		if(mission.CheckDeadline(date) && mission.IsVisible())
-			Messages::Add("You failed to meet the deadline for the mission \"" + mission.Name() + "\".",
-				Messages::Importance::Highest);
-		if(!mission.IsFailed())
-			mission.Do(Mission::DAILY, *this);
-	}
+			// Check if any special events should happen today.
+			auto it = gameEvents.begin();
+			while(it != gameEvents.end() && date >= it->GetDate())
+			{
+				GameEvent event = *it;
+				event.Apply(*this);
+				it = gameEvents.erase(it);
+			}
 
-	// Check what salaries and tribute the player receives.
-	int64_t salariesIncome = accounts.SalariesIncomeTotal();
-	int64_t tributeIncome = GetTributeTotal();
-	FleetBalance b = MaintenanceAndReturns();
-	if(salariesIncome || tributeIncome || b.assetsReturns)
-	{
-		string message = "You receive ";
-		if(salariesIncome)
-			message += Format::CreditString(salariesIncome) + " salary";
-		if(salariesIncome && tributeIncome)
-		{
-			if(b.assetsReturns)
-				message += ", ";
-			else
-				message += " and ";
+			// Check if any missions have failed because of deadlines and
+			// do any daily mission actions for those that have not failed.
+			for(Mission &mission : missions)
+			{
+				if(mission.CheckDeadline(date) && mission.IsVisible())
+					Messages::Add("You failed to meet the deadline for the mission \"" + mission.Name() + "\".",
+						Messages::Importance::Highest);
+				if(!mission.IsFailed())
+					mission.Do(Mission::DAILY, *this);
+			}
+
+			DoAccounting();
 		}
-		if(tributeIncome)
-			message += Format::CreditString(tributeIncome) + " in tribute";
-		if(salariesIncome && tributeIncome && b.assetsReturns)
-			message += ",";
-		if((salariesIncome || tributeIncome) && b.assetsReturns)
-			message += " and ";
-		if(b.assetsReturns)
-			message += Format::CreditString(b.assetsReturns) + " based on outfits and ships";
-		message += ".";
-		Messages::Add(message, Messages::Importance::High);
-		accounts.AddCredits(salariesIncome + tributeIncome + b.assetsReturns);
 	}
-
-	// For accounting, keep track of the player's net worth. This is for
-	// calculation of yearly income to determine maximum mortgage amounts.
-	int64_t assets = depreciation.Value(ships, date.DaysSinceEpoch());
-	for(const shared_ptr<Ship> &ship : ships)
-		assets += ship->Cargo().Value(system);
-
-	// Have the player pay salaries, mortgages, etc. and print a message that
-	// summarizes the payments that were made.
-	string message = accounts.Step(assets, Salaries(), b.maintenanceCosts);
-	if(!message.empty())
-		Messages::Add(message, Messages::Importance::High);
-
+	else
+	{
+		date = date + amount;
```

Comment:
I'd add a comment saying we dont fully support going back in time (as in all it does is change the date rn)
or remove the unessecary {}